packages/react/src/themes/teams/components/Loader/loaderStyles.ts

```diff
@@ -0,0 +1,85 @@
+import { LoaderProps } from '../../../../components/Loader/Loader'
+import { ComponentStyleFunctionParam, ICSSInJSStyle } from '../../../types'
+import { LoaderVariables } from './loaderVariables'
+
+export default {
+  indicator: ({
+    props: p,
+    variables: v,
+  }: ComponentStyleFunctionParam<LoaderProps, LoaderVariables>): ICSSInJSStyle => ({
+    // Reset existing styles from base theme
+    animationName: 'none',
+    animationDuration: 'unset',
+    animationIterationCount: 'unset',
+    animationTimingFunction: 'unset',
+    borderColor: 'transparent',
+    borderRadius: 0,
+    borderStyle: 'none',
+    borderWidth: 0,
+
+    height: v.containerHeights[p.size],
+    width: v.containerWidths[p.size],
+    overflow: 'hidden',
+  }),
+  svg: ({
+    props: p,
+    theme: t,
+    variables: v,
+  }: ComponentStyleFunctionParam<LoaderProps, LoaderVariables>) => {
+    const outerAnimation: ICSSInJSStyle = {
+      animationName: t.renderer.renderKeyframe(
+        () =>
+          ({
+            to: {
+              opacity: 1,
+            },
+          } as any),
+        {},
+      ),
+      animationDelay: '1.5s',
+      animationDirection: 'normal',
+      animationDuration: '.3s',
+      animationFillMode: 'both',
+      animationIterationCount: '1',
+      animationPlayState: 'running',
+      animationTimingFunction: 'ease-out',
+      display: 'block',
+      overflow: 'hidden',
+      position: 'relative',
+    }
+    const svgAnimation: ICSSInJSStyle = {
+      animationName: t.renderer.renderKeyframe(
+        () =>
+          ({
+            to: {
+              transform: `translate3d(0, ${v.svgTranslatePosition[p.size]}, 0)`,
+            },
+          } as any),
+        {},
+      ),
+      animationDelay: '0s',
+      animationDirection: 'normal',
+      animationDuration: '2s',
+      animationFillMode: 'both',
+      animationPlayState: 'running',
+      animationTimingFunction: 'steps(60, end)',
+      animationIterationCount: 'infinite',
+    }
+
+    return {
+      ...outerAnimation,
+
+      ':before': {
```

Comment:
How does this works? why we have animations for the `svg` and the `:before`? Can we somehow move this to the `indicator` and the `svg` accordingly?

Reply:
Is this because of the `svg` that is used?

Reply:
Idea with `svg` slot works fine, but [Teams loader](https://codepen.io/teams-ux-engineers/pen/1c068c6d136debb86a7164054a2c21bf) requires specific markup, i.e need one more element. Here is match:
![image](https://user-images.githubusercontent.com/14183168/55080670-2c081b00-50a7-11e9-82d8-30442bb22db1.png)

I'm opened for better idea...